### ISSUE TITLE
Pylance does not show me the docstring for the `to_csv()` method

### ISSUE THREAD

**Issue by karthiknadig** [microsoft/pylance-release#550]
@karthiknadig Thanks, I gave it a try but it does not yet look ready for prime to me. Without going into detail, Pylance does not show me the docstring for the `to_csv()` method while hovering in the following example whereas the default Python language server can do this:
```python
pd.DataFrame([
        {'file_path': html_task.output().path}
    ]).to_csv(stream))
```
Feel free to create an issue for this if Pylance should be able to support this. :-)

_Originally posted by @LinqLover in https://github.com/microsoft/vscode-python/issues/13013#issuecomment-720080563_

**Comment by gramster** [comment]
I don't think this should be a "bundled typestub bug". That should only be the case if the docstring is computed and not statically declared. In this case, the docstring is available; the method is inherited from NDFrame and the docstring is at https://github.com/pandas-dev/pandas/blob/639a9c27964b09724a0ae2acdd0e6a10e49ddd88/pandas/core/generic.py#L3228. Pylance should be finding it. I am removing the label.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.3.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202130-3-march-2021

**Comment by niderhoff** [comment]
Pylance still regularly thinks it should resolve an NDFrame while it shouldn't.
One example:

```
a = pd.DataFrame()
b = a.infer_objects()
```

b is of inferred by pyright as `NDFrame`. But if you click on "Go to Type Definition", you can see the method should return a `FrameOrSeries`:

![image](https://user-images.githubusercontent.com/5991543/152008842-e3b7f1f0-b68b-4814-bda0-915e6df013bd.png)


The type stubs (right click: "Go to Declaration"), however, will show the method should return a `NDFrame`:

![image](https://user-images.githubusercontent.com/5991543/152008131-a01bacba-791b-4915-9083-09a30227888b.png)

Now I have NDFrames all over my code where the stubs are broken basically. For example `__getitem__` of `NDFrame` is supposed to return `None` according to pyright. Why? Well, because NDFrame is implemented as an abstract class, like we can see in the pandas code:

![image](https://user-images.githubusercontent.com/5991543/152008686-564b5008-0671-41c9-b733-104642524efe.png)

I think the stubs are a not correctly generated with these abstract classes Pandas is using?
